src/daft-scan/src/anonymous.rs

```diff
@@ -69,25 +70,40 @@ impl ScanOperator for AnonymousScanOperator {
         let file_format_config = self.file_format_config.clone();
         let schema = self.schema.clone();
         let storage_config = self.storage_config.clone();
-
+        let row_groups = if let FileFormatConfig::Parquet(ParquetSourceConfig {
+            row_groups: Some(row_groups),
+            ..
+        }) = self.file_format_config.as_ref()
+        {
+            Some(row_groups.clone())
+        } else {
+            None
+        };
         // Create one ScanTask per file.
-        Ok(Box::new(files.into_iter().map(move |f| {
-            Ok(ScanTask::new(
-                vec![DataFileSource::AnonymousDataFile {
-                    path: f.to_string(),
-                    chunk_spec: None,
-                    size_bytes: None,
-                    metadata: None,
-                    partition_spec: None,
-                    statistics: None,
-                    parquet_metadata: None,
-                }],
-                file_format_config.clone(),
-                schema.clone(),
-                storage_config.clone(),
-                pushdowns.clone(),
-            )
-            .into())
-        })))
+        Ok(Box::new(files.into_iter().enumerate().map(
```

Comment:
nit: maybe slightly cleaner if we can create the ChunkSpecs as an iterator and then do a `files.into_iter().zip(chunk_specs)`?